src/na/na_cci.c

```diff
@@ -1919,6 +1923,13 @@ na_cci_progress(na_class_t * na_class, na_context_t * context,
 		hg_time_t	t1, t2;
 		cci_event_t	*event = NULL;
 
+		struct pollfd pfd;
+
+		pfd.fd = NA_CCI_PRIVATE_DATA(na_class)->fd;
+		pfd.events = POLLIN;
+
+		poll(&pfd, 1, ((int) remaining) * 1000);
```

Comment:
There is a math bug here; should be this instead:

```
poll(&pfd, 1, (int)(remaining * 1000.0));
```

The problem with the current code, is that if the value of remaining is less than 1 (remaining is a double precision floating point variable) then the cat to an integer will round it down to zero before multiplying by 1000.  We need to do the multiplication as a float before casting.